### ISSUE TITLE
Array types are not supported

### ISSUE THREAD

**Issue by robvanmieghem** [Jumpscale/go-raml#17]
http://docs.raml.org/specs/1.0/#array-types

```raml
  userview:
    properties:
        username:
            type: string
        email:
            type: string[]
            required: false
        phone:
            type: phonenumber[]
            required: false
        address:
            type: address[]
            required: false
```

generates 
```go

type Userview struct {
    Address address[] `json:"Address"`
    Email string[] `json:"Email"`
    Phone phonenumber[] `json:"Phone"`
    Username string `json:"Username"`
    
}
```

instead of 
```go

type Userview struct {
    Address []address `json:"Address"`
    Email []string `json:"Email"`
    Phone []phonenumber `json:"Phone"`
    Username string `json:"Username"`
}
```

**Comment by iwanbk** [comment]
We already have some array support, and also some test for it, but it seems buggy:

- raml : https://github.com/Jumpscale/go-raml/blob/master/commands/fixtures/struct.raml#L30

- code : https://github.com/Jumpscale/go-raml/blob/master/commands/fixtures/struct/catcat.txt

**Comment by iwanbk** [comment]
@robvanmieghem 
it is fixed in 6cf22003137b6a64e249cea15f994947a0089543 in `fix-struct-fields`, still no PR because there might be more improvement to struct support

**Comment by iwanbk** [comment]
@robvanmieghem 
fixed in #18 

there is also fix in json tag, the example above will be generated to

```
type Userview struct {
    Address []address `json:"address"`
    Email []string `json:"email"`
    Phone []phonenumber `json:"phone"`
    Username string `json:"username"`
}
```
